api/src/test/java/org/openmrs/api/PatientServiceTest.java

```diff
@@ -3308,5 +3309,65 @@ public void mergePatients_shouldMoveAllObsWithSameHierarchy() throws Exception {
 		assertEquals(1, encounterService.getEncounter(57).getObsAtTopLevel(false).size());
 		assertEquals(2, encounterService.getEncounter(57).getObsAtTopLevel(true).size());
 	}
+	@Test
+	public void addPatientMergeAction_shouldAddMergeActions() throws Exception {
+
+		PatientServiceImpl patientserviceimpl = new PatientServiceImpl();
+
+		PatientMergeAction action1 = new DummyPatientMergeAction();
+		PatientMergeAction action2 = new DummyPatientMergeAction();
+		
+
+
+		patientserviceimpl.addPatientMergeAction(action1);
+		patientserviceimpl.addPatientMergeAction(action2);
+
+		assertThat(patientserviceimpl.getPatientMergeActions().size(), is(2));
+		assertTrue(patientserviceimpl.getPatientMergeActions().contains(action1));
+		assertTrue(patientserviceimpl.getPatientMergeActions().contains(action2));
+	}
+
+	@Test
+	public void removePatientMergeAction_shouldRemoveMergeActions() throws Exception {
+
+		PatientServiceImpl patientserviceimpl = new PatientServiceImpl();
+
+		PatientMergeAction action1 = new DummyPatientMergeAction();
+		PatientMergeAction action2 = new DummyPatientMergeAction();
+
+		patientserviceimpl.addPatientMergeAction(action1);
+		patientserviceimpl.addPatientMergeAction(action2);
+
+		patientserviceimpl.removePatientMergeAction(action1);
+
+		assertThat(patientserviceimpl.getPatientMergeActions().size(), is(1));
+		assertFalse(patientserviceimpl.getPatientMergeActions().contains(action1));
+		assertTrue(patientserviceimpl.getPatientMergeActions().contains(action2));
+
+	}
+
+	@Test
+	public void test_shouldremoveMergeActionSNotFailIfActionsNotInitialized() throws Exception {
+
+		PatientServiceImpl patientserviceimpl = new PatientServiceImpl();
+
+		PatientMergeAction action = new DummyPatientMergeAction();
+		patientserviceimpl.removePatientMergeAction(action);
+
+		assertThat(patientserviceimpl.getPatientMergeActions().size(), is(0));
+	}
+
+	private class DummyPatientMergeAction implements PatientMergeAction {
```

Comment:
Cann't you use Mockito to mock `PatientMergeAction` instead of creating a dummy class?